BEIMA.Backend/MongoService/BaseDevice.cs

```diff
@@ -0,0 +1,174 @@
+            Location = new BsonDocument();
+            LastModified = new BsonDocument();
+        }
+
+        /// <summary>
+        /// Full constructor to instantiate a Device object.
+        /// </summary>
+        /// <param name="id">ObjectId of the object itself.</param>
+        /// <param name="deviceTypeId">ObjectId of the DeviceType this Device is linked to.</param>
+        /// <param name="deviceTag"></param>
+        /// <param name="manufacturer"></param>
+        /// <param name="modelNum"></param>
+        /// <param name="serialNum"></param>
+        /// <param name="yearManufactured"></param>
+        /// <param name="notes"></param>
+        public BaseDevice(ObjectId id, ObjectId deviceTypeId, string deviceTag, string manufacturer, string modelNum, string serialNum, int yearManufactured, string notes)
+        {
+            //BsonClassMap.RegisterClassMap<BaseDevice>();
+
+            Id = id;
+            DeviceTypeId = deviceTypeId;
+            DeviceTag = deviceTag ?? string.Empty;
+            Manufacturer = manufacturer ?? string.Empty;
+            ModelNum = modelNum ?? string.Empty;
+            SerialNum = serialNum ?? string.Empty;
+            YearManufactured = yearManufactured;
+            Notes = notes ?? string.Empty;
+            Fields = new BsonDocument();
+            Location = new BsonDocument();
+            LastModified = new BsonDocument();
+        }
+
+
+        [BsonId]
+        public ObjectId Id { get; set; }
+
+        [BsonElement("deviceTypeId")]
+        public ObjectId DeviceTypeId { get; set; }
+
+        [BsonElement("deviceTag")]
+        public string DeviceTag { get; set; }
+
+        [BsonElement("manufacturer")]
+        public string Manufacturer { get; set; }
+
+        [BsonElement("modelNum")]
+        public string ModelNum { get; set; }
+
+        [BsonElement("serialNum")]
+        public string SerialNum { get; set; }
+
+        [BsonElement("yearManufactured")]
+        public int YearManufactured { get; set; }
+
+        [BsonElement("notes")]
+        public string Notes { get; set; }
+
+        [BsonElement("fields")]
+        public BsonDocument Fields { get; set; }
+
+        [BsonElement("location")]
+        public BsonDocument Location { get; set; }
+
+        [BsonElement("lastModified")]
+        public BsonDocument LastModified { get; set; }
+
+        /// <summary>
+        /// Gets a BsonDocument that represents the current state of the BaseDevice object.
+        /// </summary>
+        /// <returns>BsonDocument that represents the current state of the BaseDevice object. Conforms to the schema located in BEIMA.DB.Schemas.</returns>
+        /// <exception cref="ArgumentNullException">Throws exception when any of the required fields are null.</exception>
+        public BsonDocument GetBsonDocument()
+        {
+            if (DeviceTag == null)
+            {
+                throw new ArgumentNullException($"BaseDevice - {nameof(DeviceTag)} is null");
```

Comment:
I think we can put this null check into it's own method, and pass in the field as the argument. That way we have multiple method calls, rather than almost duplicate if and throw statements